ambry-rest/src/main/java/com.github.ambry.rest/NettyResponseChannel.java

```diff
@@ -91,6 +100,11 @@
   private final Queue<Chunk> chunksToWrite = new ConcurrentLinkedQueue<Chunk>();
   private final Queue<Chunk> chunksAwaitingCallback = new ConcurrentLinkedQueue<Chunk>();
   private final AtomicLong chunksToWriteCount = new AtomicLong(0);
+  // update endpoints in following two response sets when ResponseStatus is updated.
+  private final Set<ResponseStatus> successResponseSet =
```

Comment:
these should be static. Also, this is a little hard to maintain if we add additional response codes. Perhaps a better way to do it would be to add a status code field to each `ResponseStatus` enum value.

ex:
```java
enum ResponseStatus {
  Ok(200),
  ...
  ServiceUnavailable(503),
  InsufficientCapacity(507);

  private final int statusCode;

  ResponseStatus(int statusCode) {
    this.statusCode = statusCode;
  }

  public int getStatusCode() {
    return statusCode;
  }

  boolean isSuccess() {
    return statusCode >= 200 && statusCode < 300;
  }

  boolean isRedirection() {
    return statusCode >= 300 && statusCode < 400;
  }

  boolean isClientError() {
    return statusCode >= 400 && statusCode < 500;
  }

  boolean isServerError() {
    return statusCode >= 500;
  }
}
```

Reply:
Great idea. Changes made in `ResponseStatus`.